### PR TITLE
fix(cli): always append node path when spawning child pm process

### LINKED ISSUE #2157: trying to use `bun run` with bun installed by vp causes `vp: Recursion detected but no 'node' found in PATH` error
### Describe the bug

When trying to run `bun run <script>` with bun installed by `vp i bun -g` the following error occurs

```
vp: Recursion detected but no 'node' found in PATH (excluding shims)
```
Image

discord thread https://discord.com/channels/1475973262193459293/1525793153519714424

### Reproduction

N/A

### Steps to reproduce

```bash
vp i bun -g
# using hono repo as example
git clone https://github.com/honojs/hono.git
cd hono
bun run build
```

### System Info

```shell
VITE+ - The Unified Toolchain for the Web

Environment:
  Version  25.9.0
  Source   VP_NODE_VERSION

Tool Paths:
  node  /Users/blank/.vite-plus/js_runtime/node/25.9.0/bin/node
  npm   /Users/blank/.vite-plus/js_runtime/node/25.9.0/bin/npm
  npx   /Users/blank/.vite-plus/js_runtime/node/25.9.0/bin/npx

Package Manager:
  Name          bun
  Version       1.2.20
  Source        packageManager
  Source Path   /Users/blank/Sources/hono/package.json
  Project Root  /Users/blank/Sources/hono
  Bin Path      /Users/blank/.vite-plus/package_manager/bun/1.2.20/bun/bin/bun



VITE+ - The Unified Toolchain for the Web

vp v0.2.4

Local vite-plus:
  vite-plus  Not found

Tools:
  vite             Not found
  rolldown         Not found
  vitest           Not found
  oxfmt            Not found
  oxlint           Not found
  oxlint-tsgolint  Not found
  tsdown           Not found

Environment:
  Package manager  bun v1.2.20
  Node.js          v25.9.0 (VP_NODE_VERSION)
```

### Used Package Manager

bun

### Logs

```shell

```

### Validations

- [x] Read the [Contributing Guidelines](https://github.com/voidzero-dev/vite-plus/blob/main/CONTRIBUTING.md).
- [x] Check that there isn't [already an issue](https://github.com/voidzero-dev/vite-plus/issues) for the same bug.
- [x] Confirm this is a Vite+ issue and not an upstream issue (Vite, Vitest, tsdown, Rolldown, or Oxc).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example).

### PR DESCRIPTION
**Pull request by BlankParticle**
fixes #2157
there was a gated case where in case of Bun package manager, appending node to `PATH` was being skipped, so if the child bun process tried to spawn node process `vp` would see that its running inside of another `vp` process but as there is no node path appended, it would crash

I fixed this problem by always appending node path to the child pm
the only change is removing the Bun guard case, else if related cleanup, whitespace changes and added tests

---
> I used Codex with GPT 5.6 Sol (High) to discuss and help me with setting up vite-plus locally and figure out how to write the test case

### CODE REVIEW

**Comment by jong-kyung** [comment]
@codex review

File: crates/vite_global_cli/src/shim/dispatch.rs [hunk 1]
```diff
@@ -954,66 +954,60 @@ pub async fn dispatch(tool: &str, args: &[String]) -> i32 {
...
 async fn dispatch_package_binary(tool: &str, args: &[String]) -> i32 {
-    if let Some(pm_family) = PackageManagerType::from_tool(tool) {
-        let cwd = match current_dir() {
-            Ok(path) => path,
-            Err(e) => {
-                eprintln!("vp: Failed to get current directory: {e}");
-                return 1;
-            }
-        };
+    let cwd = match current_dir() {
```
Comment (jong-kyung) [inline crates/vite_global_cli/src/shim/dispatch.rs]: Moving `current_dir()` outside `PackageManagerType::from_tool(tool)` makes every global package binary require a valid cwd before metadata lookup

Comment (BlankParticle) [reply crates/vite_global_cli/src/shim/dispatch.rs]: thanks, fixed

**Comment by liangmiQwQ** [comment]
💜 I don't find other major problems, let Codex find whether there is any other edge case.

@codex review

File: crates/vite_cli_snapshots/tests/cli_snapshots/fixtures/shim_bun_runs_node_child/snapshots.toml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[[case]]
+name = "shim_bun_runs_node_child"
+vp = "global"
+skip-platforms = ["windows"]
```
Comment (liangmiQwQ) [inline crates/vite_cli_snapshots/tests/cli_snapshots/fixtures/shim_bun_runs_node_child/snapshots.toml]: Maybe we can allow this test to run on Windows?

Comment (BlankParticle) [reply crates/vite_cli_snapshots/tests/cli_snapshots/fixtures/shim_bun_runs_node_child/snapshots.toml]: i am actually not too sure about this, codex suggested me this template, maybe windows will work too

File: crates/vite_global_cli/src/shim/dispatch.rs [hunk 3]
```diff
@@ -965,36 +965,32 @@ async fn dispatch_package_binary(tool: &str, args: &[String]) -> i32 {
...
                                 }
                             }
-                            Err(e) => {
-                                eprintln!("vp: Failed to install Node {}: {e}", node_version);
-                                return 1;
-                            }
+                        }
+                        Err(e) => {
+                            eprintln!("vp: Failed to install Node {}: {e}", node_version);
+                            return 1;
```
[the remark this answers is not in the document]
Comment (liangmiQwQ) [reply crates/vite_global_cli/src/shim/dispatch.rs:993]: Running a native Bun does not require Node.js setup. What if adjusting this PR's direction to stop injecting `VP_TOOL_RECURSION` for Bun? Would that be better?

[the remark this answers is not in the document]
Comment (BlankParticle) [reply crates/vite_global_cli/src/shim/dispatch.rs:993]: i think that wouldn't be best idea, bun might spawn a node process at any time, like launching a vite dev server
we can't just add bypasses for bun and any other issues in future with other runtime

[the remark this answers is not in the document]
Comment (liangmiQwQ) [reply crates/vite_global_cli/src/shim/dispatch.rs:993]: Okay, make sense. Downloading a runtime not a big deal but help make the behavior consistent. Let's keep it.

**Review by liangmiQwQ — APPROVED** [review]
Since the other team members may be busy these days, I can’t make sure when your PR will be merged. Thanks for your contribution 💯

**Review by wan9chi — APPROVED** [review]
